File: chapter_002/bmi.py
#!/bin/bash/env python3

height_cm = float(input("身長(cm): "))
weight = float(input("体重(kg): "))

bmi = weight / (height_cm / 100) ** 2
print(f"BMI: {bmi:.2f}")

if bmi < 18.5:
    print("低体重")
elif bmi < 25:
    print("標準")
elif bmi < 30:
    print("肥満(1度)")
elif bmi < 35:
    print("肥満(2度)")
elif bmi < 40:
    print("肥満(3度)")
else:
    print("肥満(4度)")

File: chapter_002/bmi.rs
fn input(prompt: &str) -> f64 {
    println!("{}", prompt);
    let mut input = String::new();
    std::io::stdin().read_line(&mut input).expect("Failed to read line");
    input.trim().parse::<f64>().expect("Failed to parse input")
}

fn main() {
    let height = input("Height: ");
    let weight = input("Weight: ");

    let bmi = weight / ((height/ 100.0).powf(2.0));
    println!("BMI: {:.2}", bmi);

    if bmi < 18.5 {
        println!("Underweight");
    } else if bmi < 25.0 {
        println!("Normal");
    } else if bmi < 30.0 {
        println!("Overweight (1)");
    } else if bmi < 35.0 {
        println!("Overweight (2)");
    } else if bmi < 40.0 {
        println!("Overweight (3)");
    } else {
        println!("Obese");
    }
}
